target/linux/ar71xx/base-files/lib/ar71xx.sh

```diff
@@ -581,8 +581,18 @@ ar71xx_board_detect() {
 		name="cpe505n"
 		;;
 	*"CPE510/520")
-		name="cpe510"
-		tplink_pharos_board_detect "$(tplink_pharos_get_model_string | tr -d '\r')"
+                name="cpe510"
+                tplink_v2_check() {
+                        local part
+                        part=$(find_mtd_part 'product-info')
+                        dd if=$part bs=1 skip=4360 count=64 2>/dev/null | tr -d '\r\0' | head -n 1
+                        }
+                        md_string="$(tplink_v2_check)"
+                        if [ "${md_string:35:3}" == "2.0" ] || [ "${md_string:26:3}" == "2.0" ] ; then
```

Comment:
Does this (and the required tplink_pharos_v2_get_model_string ) work with a CPE510v1?